### PR TITLE
Add cancelled workshop status

### PR DESCRIPTION
**Pull request by krzys-h**

### CODE REVIEW

File: templates/program.html [hunk 1]
```diff
@@ -81,13 +81,13 @@ <h1>{{ title }}</h1>
             {% for workshop, registered in workshops %}
             <div class="panel panel-default" id="{{ workshop.name }}">
                 <div class="panel-heading" style="font-size: 16pt;">
-                    {% if workshop.page_content_is_public %}<a href="{% url 'workshop_page' workshop.name %}"><big>{{ workshop.title }}</big></a>{% else %}<big>{{ workshop.title }}</big>{% endif %}
+                    {% if workshop.page_content_is_public %}{% if workshop.status == 'X' %}(ODWOŁANE) {% endif %}<a href="{% url 'workshop_page' workshop.name %}"><span style="font-size: larger;">{{ workshop.title }}</span></a>{% else %}<span style="font-size: larger;">{{ workshop.title }}</span>{% endif %}
```
Comment (cytadela8) [inline templates/program.html]: - Uses classes from bootstrap for font size.
- Split into multiple lines.

File: templates/workshop.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <article class="col-sm-12 maincontent">
         <h1>{{ title }}</h1>
         {% if has_perm_to_edit %}
-            {% if workshop.status == 'Z' %}
+            {% if workshop.status == 'Z' or workshop.status == 'X' %}
```
Comment (cytadela8) [inline templates/workshop.html]: Add some kind of function to workshop for this check. For example `is_accepted`

File: templates/program.html [hunk 3]
```diff
@@ -81,13 +81,17 @@ <h1>{{ title }}</h1>
...
                 <div class="panel-heading" style="font-size: 16pt;">
-                    {% if workshop.page_content_is_public %}<a href="{% url 'workshop_page' workshop.name %}"><big>{{ workshop.title }}</big></a>{% else %}<big>{{ workshop.title }}</big>{% endif %}
+                    {% if workshop.page_content_is_public %}
+                        {% if workshop.status == 'X' %}(ODWOŁANE) {% endif %}<a href="{% url 'workshop_page' workshop.name %}"><span style="font-size: larger;">{{ workshop.title }}</span></a>
+                    {% else %}
+                        <span style="font-size: larger;">{{ workshop.title }}</span>
+                    {% endif %}
                         <span class="pull-right">
-                            <big>
+                            <span style="font-size: larger;">
```
Comment (cytadela8) [inline templates/program.html]: please add new class

File: templates/program.html [hunk 4]
```diff
@@ -81,13 +81,17 @@ <h1>{{ title }}</h1>
             {% for workshop, registered in workshops %}
             <div class="panel panel-default" id="{{ workshop.name }}">
                 <div class="panel-heading" style="font-size: 16pt;">
-                    {% if workshop.page_content_is_public %}<a href="{% url 'workshop_page' workshop.name %}"><big>{{ workshop.title }}</big></a>{% else %}<big>{{ workshop.title }}</big>{% endif %}
+                    {% if workshop.page_content_is_public %}
+                        {% if workshop.status == 'X' %}(ODWOŁANE) {% endif %}<a href="{% url 'workshop_page' workshop.name %}"><span class="big">{{ workshop.title }}</span></a>
+                    {% else %}
+                        <span class="big">{{ workshop.title }}</span>
+                    {% endif %}
                         <span class="pull-right">
```
Comment (cytadela8) [inline templates/program.html]: Move class here and remove addictional span.

File: templates/workshop.html [hunk 5]
```diff
@@ -6,8 +6,11 @@
 <div class="container">
     <article class="col-sm-12 maincontent">
         <h1>{{ title }}</h1>
+        {% if workshop.is_publicly_visible %}
+            {% include '_workshopeditheader.html' %}
+        {% endif %}
         {% if has_perm_to_edit %}
-            {% if workshop.status == 'Z' %}
+            {% if workshop.status == 'Z' or workshop.status == 'X' %}
```
Comment (cytadela8) [inline templates/workshop.html]: `is_publicly_visible` should be used here

File: wwwapp/models.py [hunk 6]
```diff
@@ -244,6 +244,9 @@ def qualified_count(self):
             return None
         return self.workshopparticipant_set.filter(qualification_result__gte=self.qualification_threshold).count()
 
+    def is_publicly_visible(self):  # Zaakceptowane lub odwołane
```
Comment (grzegorz225) [inline wwwapp/models.py]: Add documentation and return type

File: templates/program.html [hunk 7]
```diff
@@ -81,13 +81,17 @@ <h1>{{ title }}</h1>
             {% for workshop, registered in workshops %}
             <div class="panel panel-default" id="{{ workshop.name }}">
                 <div class="panel-heading" style="font-size: 16pt;">
-                    {% if workshop.page_content_is_public %}<a href="{% url 'workshop_page' workshop.name %}"><big>{{ workshop.title }}</big></a>{% else %}<big>{{ workshop.title }}</big>{% endif %}
+                    {% if workshop.page_content_is_public %}
+                        {% if workshop.status == 'X' %}(ODWOŁANE) {% endif %}<a href="{% url 'workshop_page' workshop.name %}"><span class="big">{{ workshop.title }}</span></a>
```
Comment (grzegorz225) [inline templates/program.html]: Split line

File: wwwapp/models.py [hunk 8]
```diff
@@ -244,7 +244,10 @@ def qualified_count(self):
             return None
         return self.workshopparticipant_set.filter(qualification_result__gte=self.qualification_threshold).count()
 
-    def is_publicly_visible(self):  # Zaakceptowane lub odwołane
+    """
+    Czy warsztaty powinny być publicznie widoczne (są zaakceptowane lub odwołane)
```
Comment (grzegorz225) [inline wwwapp/models.py]: Don't use polish in documentation

**Pull request merged**